### ISSUE TITLE
[java] IdenticalCatchBranches false negative with different methods eventually throwing same exception

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#4106]
**Rule:** [IdenticalCatchBranches](https://pmd.github.io/latest/pmd_rules_java_codestyle.html#identicalcatchbranches)

Hi, this fix just take method call out of consideration.

```java
class Foo {
                static {
                    try {
                        // do something
                    } catch (Exception1 exception) {
                        exception(exception);
                    } catch (Exception2 exception2) {
                        exception2(exception2);
                    }
                }
                private static void exception(Exception exception) {
                    throw new Exception3("Error message 1", exception);
                }
                private static exception2(Exception exception) {
                    // instead of: throw new Exception3("Error message 2", exception);
                    throw new Exception3("Error message 1", exception);
                }
            }
```

If I change the second test case like this, PMD will not hint, which may be a false negative.

_Originally posted by @Han-Cui in https://github.com/pmd/pmd/issues/1161#issuecomment-1229156441_

**Comment by Shivangip285** [comment]
instanceof can be used in fixing this.

**Comment by oowekyala** [comment]
I don't think we should pursue this direction. It would be bad for performance, and this is the kind of duplication that CPD or other copy-paste detection tools are better suited in detecting.
